### PR TITLE
[SPARK-22713][CORE][TEST][FOLLOWUP] Fix flaky ExternalAppendOnlyMapSuite due to timeout

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
## What changes were proposed in this pull request?

SPARK-22713 uses [`eventually` with the default timeout `150ms`](https://github.com/apache/spark/pull/21369/files#diff-5bbb6a931b7e4d6a31e4938f51935682R462). It causes flakiness because it's executed once when GC is slow.

```scala
eventually {
  System.gc()
  ...
}
```

**Failures**
```scala
org.scalatest.exceptions.TestFailedDueToTimeoutException:
The code passed to eventually never returned normally.
Attempted 1 times over 501.22261 milliseconds.
Last failure message: tmpIsNull was false.
```
- master-test-sbt-hadoop-2.7
  [4916](https://amplab.cs.berkeley.edu/jenkins/view/Spark%20QA%20Test%20(Dashboard)/job/spark-master-test-sbt-hadoop-2.7/4916)
  [4907](https://amplab.cs.berkeley.edu/jenkins/view/Spark%20QA%20Test%20(Dashboard)/job/spark-master-test-sbt-hadoop-2.7/4907)
  [4906](https://amplab.cs.berkeley.edu/jenkins/view/Spark%20QA%20Test%20(Dashboard)/job/spark-master-test-sbt-hadoop-2.7/4906)

- spark-master-test-sbt-hadoop-2.6
  [4979](https://amplab.cs.berkeley.edu/jenkins/view/Spark%20QA%20Test%20(Dashboard)/job/spark-master-test-sbt-hadoop-2.6/4979)
  [4974](https://amplab.cs.berkeley.edu/jenkins/view/Spark%20QA%20Test%20(Dashboard)/job/spark-master-test-sbt-hadoop-2.6/4974)
  [4967](https://amplab.cs.berkeley.edu/jenkins/view/Spark%20QA%20Test%20(Dashboard)/job/spark-master-test-sbt-hadoop-2.6/4967)
  [4966](https://amplab.cs.berkeley.edu/jenkins/view/Spark%20QA%20Test%20(Dashboard)/job/spark-master-test-sbt-hadoop-2.6/4966)

## How was this patch tested?

Pass the Jenkins.

### CODE REVIEW

File: core/src/test/scala/org/apache/spark/util/collection/ExternalAppendOnlyMapSuite.scala [hunk 1]
```diff
@@ -457,7 +458,7 @@ class ExternalAppendOnlyMapSuite extends SparkFunSuite
     // https://github.com/scala/scala/blob/2.13.x/test/junit/scala/tools/testing/AssertUtil.scala
     // (lines 69-89)
     // assert(map.currentMap == null)
-    eventually {
+    eventually(timeout(5 seconds), interval(200 milliseconds)) {
```
Comment (eyalfa) [inline core/src/test/scala/org/apache/spark/util/collection/ExternalAppendOnlyMapSuite.scala]: I think best practice to encounter this is specifying config patience in sbt's test options.
having that said, I'v once had to 'grant' a longer timeout to a specific class so I've achieved this by overriding the `spanScalaeFactor` method

`override  def spanScaleFactor: Double = super.spanScaleFactor * 3`

please notice that there's another usage of `eventually` in line 519, this one 'manually' waits 500 millis before testing which might explain why you didn't see it failing in CI, looking at it now it seems like a bad practice since `eventually` is designed to control both the timeout and the intervals between trying. 

to summarize: best practice is to control this in sbt's test settings, if needed you can further control it in a specific class, in any case you have to make sure you handle all invocations of `eventually` (which is easier and less error prone by leveraging scalaTest's mechanisms).

**Review by eyalfa — CHANGES_REQUESTED** [review]
there's another eventuall call site in the class, please consider my suggestion to fix all call sites without looking them up one by one.

File: core/src/test/scala/org/apache/spark/util/collection/ExternalAppendOnlyMapSuite.scala (hunk 1 above)
Comment (eyalfa) [reply core/src/test/scala/org/apache/spark/util/collection/ExternalAppendOnlyMapSuite.scala]: @dongjoon-hyun, thanks for cleaning up my mess! :sunglasses:

Comment (dongjoon-hyun) [reply core/src/test/scala/org/apache/spark/util/collection/ExternalAppendOnlyMapSuite.scala]: Thank you for review, @eyalfa .

If you are looking at Spark code, you must notice that the most test cases are designed to have their suitable timeout and interval. We don't use a long global value for all test cases. That will hide potential big issues.

Also, I already knew that the other instance you mentioned, and 4 instance more on `ProcessingTimeExecutorSuite`. However, I didn't want to change it because I usually don't touch the test cases if it's not flaky.

Specifically, this instance is really serious in our Jenkins environment. The above 7 failures are only recent samples from 2 test series. We have additional 3 test series on branch `master`; maven-hadoop2.6, maven-hadoop2.7, sbt-scala-2.12. For `branch-2.4`, we have 4 more.

Comment (dongjoon-hyun) [reply core/src/test/scala/org/apache/spark/util/collection/ExternalAppendOnlyMapSuite.scala]: This is a tiny issue annoying us. It's not a mess. :) Thank you again for your quick response.

Comment (cloud-fan) [reply core/src/test/scala/org/apache/spark/util/collection/ExternalAppendOnlyMapSuite.scala]: +1 for not overkilling it. If this is the only flaky test, let's fix this instance

**Comment by cloud-fan** [comment]
retest this please

**Review by HyukjinKwon — APPROVED** [review]

**Comment by cloud-fan** [comment]
thanks, merging to master/2.4!

**Comment by dongjoon-hyun** [comment]
Thank you, @cloud-fan and @HyukjinKwon.
